src/box/lua/config/applier/lua.lua

```diff
@@ -0,0 +1,32 @@
+local alloc = require('internal.alloc')
+local log = require('internal.config.utils.log')
+
+local is_lua_memory_applied = false
+
+local function apply(config)
+    local configdata = config._configdata
+    local memory = configdata:get('lua.memory', {use_default = true})
+
+    if memory == alloc.getmemory() then
+        return
+    end
+
+    if not is_lua_memory_applied then
+        -- The call throws an error if the user tries to limit the
+        -- memory with values < 256MB or with an amount less than
+        -- currently allocated.
+        alloc.setmemory(memory)
+        log.verbose(('lua.apply: set memory limit to %d'):format(memory))
+
+        is_lua_memory_applied = true
+    else
+        local warning = 'lua.apply: non-dynamic option lua.memory ' ..
+                        'will not be set until the instance is restarted'
+        config._aboard:set({type = 'warn', message = warning})
```

Comment:
Technically this code is called once on reconfiguration and at most twice on startup (and the first call can't issue the alert). However, I wouldn't bother much about a code on the upper level and just be ready to several calls per (re)configuration process. IOW, I would add a key for this alert, so several alerts with this message can't be issued by construction.

Reply:
I've added the key for the alert. Now it's impossible for the alert to be duplicated. Thank you.